omnigibson/prims/cloth_prim.py

```diff
@@ -146,27 +146,54 @@ def kinematic_only(self):
         """
         return False
 
-    def _compute_particle_positions(self, keypoints_only=False):
+    def compute_particle_positions(self, idxs=None):
         """
         Compute individual particle positions for this cloth prim
 
         Args:
-            keypoints_only (bool): If True, will only return the keypoint particle state
+            idxs (n-array or None): If set, will only calculate the requested indexed particle state
 
         Returns:
             np.array: (N, 3) numpy array, where each of the N particles' positions are expressed in (x,y,z)
                 cartesian coordinates relative to the world frame
         """
-        r = T.quat2mat(self.get_orientation())
-        t = self.get_position()
+        t, r = self.get_position_orientation()
+        r = T.quat2mat(r)
         s = self.scale
 
-        p_local = np.array(self.get_attribute(attr="points"))
-        p_local = p_local[self._keypoint_idx] if keypoints_only else p_local
+        # Don't copy to save compute, since we won't be returning a reference to the underlying object anyways
+        p_local = np.array(self.get_attribute(attr="points"), copy=False)
+        p_local = p_local[idxs] if idxs is not None else p_local
         p_world = (r @ (p_local * s).T).T + t
 
         return p_world
 
+    def set_particle_positions(self, positions, idxs=None):
+        """
+        Sets individual particle positions for this cloth prim
+
+        Args:
+            positions (n-array): (N, 3) numpy array, where each of the N particles' positions are expressed in (x,y,z)
+                cartesian coordinates relative to the world frame
+            idxs (n-array or None): If set, will only set the requested indexed particle state
+        """
+        n_expected = self._n_particles if idxs is None else len(idxs)
+        assert len(positions) == n_expected, \
+            f"Got mismatch in particle setting size: {len(positions)}, vs. number of expected particles {n_expected}!"
+
+        r = T.quat2mat(self.get_orientation())
+        t = self.get_position()
+        s = self.scale
+        p_local = (r.T @ (positions - t).T).T / s
+
+        # Fill the idxs if requested
+        if idxs is not None:
+            p_local_old = np.array(self.get_attribute(attr="points"))
+            p_local_old[idxs] = p_local
+            p_local = p_local_old
```

Comment:
nit, would the following work (directly modify the array in place)?
```
if idxs is not None:
    p_local_old = np.array(self.get_attribute(attr="points"), copy=False)
    p_local_old[idxs] = p_local
```

Reply:
actually, as it turns out we can't do this )): if `copy=False` the array becomes read-only, and we can't overwrite indices